src/DynamoCoreWpf/ViewModels/Search/NodeAutoCompleteSearchViewModel.cs

```diff
@@ -52,42 +42,240 @@ internal void PopulateAutoCompleteCandidates()
+            {
+                this.typeNameToCompareTo = typeNameToCompareTo;
+                this.core = core;
+            }
+
+            public int Compare(NodeSearchElement x, NodeSearchElement y)
+            {
+                return CompareNodeSearchElementByTypeDistance(x, y, typeNameToCompareTo, core);
+            }
+
+
+            /// <summary>
+            /// Compares two nodeSearchElements - general rules of the sort as follows:
+            /// If all return types of the two searchElements are the same, they are equal.
+            /// If any return type of both searchElements is an exact match for our input type, they are equal.
+            /// If a searchElement is null, it is larger than the other element.
+            /// If a single searchElement's return type list contains an exact match it is smaller.
+            /// If the minimum type distance between a searchElements return types and our inputType than this searchElement is smaller. (closer)
+            /// If the minimim type distances are the same the searchElements are equal.
+            /// </summary>
+            /// <param name="x"></param>
+            /// <param name="y"></param>
+            /// <param name="typeNameToCompareTo"></param>
+            /// <param name="core"></param>
+            /// <returns></returns>
+            private int CompareNodeSearchElementByTypeDistance(NodeSearchElement x, NodeSearchElement y, string typeNameToCompareTo, ProtoCore.Core core)
+            {
+                /*
+                 * -1	x preceeds y.
+                 0 x and y are equal
+                 1	x is after y.
+                 */
+                var none = new string[] { Resources.NoneString };
+                var xTypeNames = x.OutputParameters;
+                var yTypeNames = y.OutputParameters;
+
+                //if either element is a ztsearchElement then just use the type name (this strips off the rank)
+                if (x is ZeroTouchSearchElement xzt)
+                {
+                    xTypeNames = new string[] { xzt.Descriptor.ReturnType.Name };
+                }
+                else
+                {
+                    //if inputPortType is an array, lets just use the class name to match 
+                    xTypeNames = xTypeNames.Select(type => ParserUtils.ParseWithCore($"dummyName:{ type};", core).CodeBlockNode.Children().ElementAt(0).Name);
+                }
+                if (y is ZeroTouchSearchElement yzt)
+                {
+                    yTypeNames = new string[] { yzt.Descriptor.ReturnType.Name };
+                }
+                else
+                {
+                    yTypeNames = yTypeNames.Select(type => ParserUtils.ParseWithCore($"dummyName:{ type};", core).CodeBlockNode.Children().ElementAt(0).Name);
+
+                }
+
+                if (xTypeNames.SequenceEqual(yTypeNames))
+                {
+                    return 0;
+                }
```

Comment:
When we say multiple x types and y types, are we future-proofing for multi-output nodes?

Reply:
yes, I did this just to future proof, it doesn't handle zeroTouch nodes, but it *should* handle nodeModel nodes that return multiple values. I think that depends on if the types are actually specified as attributes on the nodeModel class.